tutorials/lesson3-anatomy/coin.template.qnt

```diff
@@ -0,0 +1,953 @@
+   the map `balances`, REPL would show a runtime error.
+
+ - `balances.set(receiver, newBal)` produces a new map, that assigns
+   the value of `newBal` to the key `receiver`, and keeps the other key-value
+   pairs as in `balances`.
+
+Now it's time to mint some coins in REPL! Try the following:
+
+            <run target="markdown">echo 'init\n mint(minter, "bob", 2023)\n state' | quint -r coin.qnt::Coin</run>
+            <run target="codetour">echo 'init\n mint(minter, "bob", 2023)\n state' | quint -r ./lesson3-anatomy/coin.qnt::Coin</run>
+
+            <block>
+As you can see, we can mint coins for Bob.
+
+**Exercise:** Mint more coins for Bob, actually, mint `MAX_UINT` coins.
+Do you understand what happened in this case?
+            </block>
+
+        </description>
+        <file>$file</file>
+        <line>$line</line>
+      </step>
+    !*/
+
+    // Sends an amount of existing coins from any caller (sender)
+    // to a receiver's address.
+    // Note that we have to add `sender` as an action parameter,
+    // which is accessed via implicit `msg.sender` in Solidity.
+    action send(sender: Addr, receiver: Addr, amount: UInt): bool = all {
+        require(not(amount > balances.get(sender))),
+        if (sender == receiver) {
+            balances' = balances
+        } else {
+            val newSenderBal = balances.get(sender) - amount
+            val newReceiverBal = balances.get(receiver) + amount
+            all {
+                // Again, Solidity does an automatic overflow test.
+                // We do it ourselves.
+                require(isUInt(newSenderBal)),
+                require(isUInt(newReceiverBal)),
+                balances' =
+                    balances
+                    .set(sender, newSenderBal)
+                    .set(receiver, newReceiverBal)
+            }
+        },
+        // keep the minter unchanged
+        minter' = minter,
+    }
+    /*!
+      <step>
+        <title>Defining the action `send`</title>
+        <description>
+
+If you understood the mechanics of the action `mint`, you should easily
+figure out the behavior of `send`.
+
+Play with `mint` and `send` in REPL! The simplest scenario would be:
+
+            <run target="markdown">echo 'init\n mint(minter, "bob", 2023)\n send("bob", "eve", 1024)\n state' | quint -r coin.qnt::Coin</run>
+            <run target="codetour">echo 'init\n mint(minter, "bob", 2023)\n send("bob", "eve", 1024)\n state' | quint -r ./lesson3-anatomy/coin.qnt::Coin</run>
+
+        </description>
+        <file>$file</file>
+        <line>$line</line>
+      </step>
+    !*/
+
+    // All possible behaviors of the protocol in one action.
+    // Note that random simulation produces valid inputs to
+    // this action with a low probability, since amount are chosen at random.
+    // A symbolic model checker such as Apalache does not have any problem
+    // with this. Random simulation requires a more restricted version of step.
```

Comment:
Maybe these should be forward pointers to the sections on REPL test/runs and Apalache?
In general, I think using the name of a Quint feature instead of general academic terms, will make it easier for users to connect and navigate.

Reply:
I was actually thinking of referring the reader from the REPL tutorial to this one, if they wanted to learn about runs. The exposition here is already good. We don't have to repeat it in the REPL tutorial.

Reply:
The comment above is not even true, as the probability happened ok, according to my experiments :))

Removed the comment.